tests/js/client/shell/shell-debug-dump.js

```diff
@@ -0,0 +1,259 @@
+const vn3 = "testVertex3";
+const cn1 = "edgeTestCollection";
+const cn2 = "edgeTestCollection2";
+const relationName = "isRelated";
+const graphName = "myGraph";
+const dbName = "testDatabase";
+const fileName = fs.getTempFile() + "-debugDump";
+const outFileName = fs.getTempFile() + "-inspectDump";
+
+
+function recreateEmptyDatabase () {
+  db._useDatabase("_system");
+  db._dropDatabase(dbName);
+  db._createDatabase(dbName);
+  db._useDatabase(dbName);
+  try {
+    db._graphs.document(graphName);
+    fail();
+  } catch (err) {
+    assertEqual(err.errorMessage, "document not found");
+  }
+}
+
+function createOrderedKeysArray (collName) {
+  let keys = db[collName].toArray().map(doc => doc._key);
+  keys.sort();
+  return keys;
+}
+
+function testGeneralWithSmartGraphNamed (amountExamples) {
+  const graphBefore = db._graphs.document(graphName);
+  delete graphBefore._rev;
+  const query = `FOR v,e, p IN 0..10 ANY 'customer/customer-abc' GRAPH '${graphName}'  RETURN {key: e._key}`;
+  explainer.debugDump(fileName, query, {}, {examples: amountExamples});
+  explainer.inspectDump(fileName, outFileName);
+  recreateEmptyDatabase();
+  internal.load(outFileName);
+  assertNull(db[vn1].any());
+  assertNull(db[vn2].any());
+  assertNull(db[relationName].any());
+  const graphAfter = db._graphs.document(graphName);
+  delete graphAfter._rev;
+  assertEqual(graphBefore, graphAfter, "graph recreated after debugDump is not the same as the original");
+}
+
+function testGeneralWithGraphNamed (amountExamples) {
+  const graphBefore = db._graphs.document(graphName);
+  delete graphBefore._rev;
+  const query = `FOR v,e, p IN 0..10 ANY  '${vn1}/value-1' GRAPH '${graphName}'  RETURN {vertex: p}`;
+  explainer.debugDump(fileName, query, {}, {examples: amountExamples});
+  explainer.inspectDump(fileName, outFileName);
+  recreateEmptyDatabase();
+  internal.load(outFileName);
+  const graphAfter = db._graphs.document(graphName);
+  delete graphAfter._rev;
+  assertEqual(graphBefore, graphAfter, "graph recreated after debugDump is not the same as the original");
+}
+
+function testGeneralWithGraphNoNameInQuery (amountExamples) {
+  const collNameBefore = db[cn1].name();
+  const query = `WITH ${vn1} FOR v,e, p IN 0..10 ANY  '${vn1}/value-1' ${cn1} RETURN {from: e._from, to: e._to}`;
+  explainer.debugDump(fileName, query, {}, {examples: amountExamples});
+  explainer.inspectDump(fileName, outFileName);
+  recreateEmptyDatabase();
+  internal.load(outFileName);
+  const collNameAfter = db[cn1].name();
+  assertEqual(collNameBefore, collNameAfter);
+}
+
+function debugDumpSmartGraphTestsuite() {
+  return {
+    setUp: function () {
+      db._createDatabase(dbName);
+      db._useDatabase(dbName);
+      assertNotNull(smartGraph);
+      const rel = smartGraph._relation(relationName, [vn1], [vn2]);
+      const namedGraph = smartGraph._create(graphName, [rel], [], {
+        smartGraphAttribute: "value",
+        isDisjoint: true,
```

Comment:
Can we duplicate this test, one with isDisjoint: true, one with isDisjoint: false?
I do not really expect a red test here, but the layout of a disjoint edge collecion differs from a "normal" one.

Reply:
Ok